### ISSUE TITLE
Automated migration of all non-GIFT datasets we plan to keep

### ISSUE THREAD

**Issue by sglavoie** [openspending/openspending#1488]
**Context:**
* Part of Epic #1479.
* GIFT datasets were backed up as part of #1486.

**Job story:**

* When we are about to migrate all OpenSpending datasets we want to keep, we want to make sure that we have a solid plan to execute successfully the migration so we can confidently know that all OpenSpending data will be backed up to Google Storage and GitHub and that nothing we need will be mistakenly left behind.

**Technical job stories:**
* When I run the automated migration, I want to back up a specific list of (pre-selected) datasets from S3 to Google Storage without having to intervene manually so I can let the script run in the background and be confident that everything is correctly backed up when I see the logs for the operations being performed.
* When I execute the migration script, I also want to make sure that if any error arises during the process (mostly due to network issues), the script can be successfully relaunched and will continue working from the last "save point" so I can perform the migration as seamlessly as possible without wondering if data is missing on GitHub or Google Storage.

## Acceptance Criteria

* [x] All the (non-GIFT) datasets we want to keep have been backed up at [github.com/os-data](https://github.com/os-data) with Git-LFS and all the data is accessible on a Google Storage bucket.

## Tasks

* [x] **(quick)** Review the [spreadsheet containing the filtered datasets to keep](https://docs.google.com/spreadsheets/d/1Cno0jUkfl8ozf0qjaNhuCmJEmMaL2U65sNONqVTEEnM/edit#gid=402256245) to make sure it doesn't contain unneeded data (tests or duplicates → Sort `Title` column alphabetically and filter with the `Keep` column: there are not many datasets to watch out for).
  * [x] Make sure all filters listed in #1482 were properly applied.
  * [x] Double-check that there are no GIFT datasets being backed up again by comparing with https://github.com/gift-data (look for the dataset IDs from the script used to extract them).
* [x] Get admin access to https://github.com/os-data to create the necessary repositories.
  * [x] Use GitHub API to create the datasets (refer to the example in the guide with `curl`).
  * [x] Push all datasets with Git-LFS to Google Cloud Storage.
  * [x] Push all metadata to GitHub repositories.
* [x] Set up a local directory following the instructions from the guide [Migrating data from S3 to Google Cloud Storage using Giftless](https://github.com/openspending/audit-for-migration-2020/blob/master/s3_to_google_cloud_storage/README.md).
  * [x] Create a Google Storage bucket in a new project (e.g. "OpenSpending Datasets").
    * [x] Configure Giftless to use the necessary project ID, private key and client ID from that new project on Google Storage.
  * [x] Update the code from [retrieve_resources.py](https://github.com/openspending/audit-for-migration-2020/blob/master/s3_to_google_cloud_storage/retrieve_resources.py) to get the list of datasets from an external file (~~CSV should be convenient by just copying/pasting the desired selection from [the spreadsheet](https://docs.google.com/spreadsheets/d/1Cno0jUkfl8ozf0qjaNhuCmJEmMaL2U65sNONqVTEEnM/edit#gid=402256245) and converting tabs to commas~~ → went for JSON to build a dictionary in one step).
    * [x] Have a function to generate a proper title for the dataset repository name on GitHub and create them all in one go.
    * [x] Have script to create all datasets on GitHub.
  * [x] Automate the push of all the datasets with Git-LFS (convert [this Bash script](https://github.com/openspending/audit-for-migration-2020/blob/master/s3_to_google_cloud_storage/push_lfs.sh) to a more scalable implementation).
  * [x] Automate the step of pushing to GitHub by tweaking [this other Bash script](https://github.com/openspending/audit-for-migration-2020/blob/master/s3_to_google_cloud_storage/push_git.sh).
* [x] Retrieve all `datapackage.json` files.
* [x] Retrieve all resources files.

/cc @rufuspollock

**Comment by sglavoie** [comment]
**TL;DR:** done!

**Quick update**:

* There are 76 new unique datasets that have been considered for a backup weighting ~6.3GB (apart from the datasets we already have for GIFT which are ~8.4GB in size).
  * All of them have been backed up, with their respective `datapackage.json` and all accompanying resources: they are located on Google Cloud Storage and each dataset has a repository at https://github.com/os-data.
  * The list of datasets that have been backed up can be looked up quickly [in this spreadsheet](https://docs.google.com/spreadsheets/d/1Cno0jUkfl8ozf0qjaNhuCmJEmMaL2U65sNONqVTEEnM/edit#gid=402256245), filtering column `E` ("Keep?") for `Yes` values only.
    * GIFT datasets were also backed up previously [in this repo](https://github.com/gift-data) and can be looked up in the same spreadsheet by filtering the column `F` ("Notes") and showing the rows containing the value "GIFT dataset: already backed up" (column `E` should display all `No` values for GIFT datasets to show up since they were not considered in this particular backup).
* The scripts used to do the migrations were committed here https://github.com/openspending/audit-for-migration-2020/commit/f5b2bae40dc34035192f5ee61c857266d9e054a4. (Note: the Bash scripts are far from being elegant, but they worked in one go :sunglasses:.)

![image](https://user-images.githubusercontent.com/32682903/95381843-61054880-08ae-11eb-84bc-3f37058aea8c.png)
